ci-operator/step-registry/cucushift/installer/check/vsphere/template/cucushift-installer-check-vsphere-template-commands.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+source "${SHARED_DIR}/govc.sh"
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+# check template in cpms and machineset
+check_result=0
+template_config=$(yq-go r "${SHARED_DIR}/install-config.yaml" 'platform.vsphere.failureDomains[*].topology.template')
+template_config=${template_config##*/}
+template_ms=$(oc get machineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.spec.providerSpec.value.template')
+template_ms=${template_ms##*/}
+if [[ ${template_config} != "${template_ms}" ]]; then
+    echo "ERROR: template specify in install-config is ${template_config},  not same as machineset's template ${template_ms}. please check"
+    check_result=1
+else 
+    echo "INFO template specify in install-config is ${template_config}, same as machineset's template ${template_ms}. check successful "
+fi
+
+ocp_minor_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f2)
+if (( ${ocp_minor_version} > 15 )); then
+	template_cpms=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.template')
```

Comment:
for OCP 4.16+, the vsphere cmps becomes GA feature now?

Reply:
yes

Reply:
If that, we should update `cucushift-installer-check-control-plane-machinesets` together?

Reply:
If it becomes a GA feature, `v1beta1` is still being used, is that expected?